### PR TITLE
Fix Fullscreen (No Answer Buttons) Reviewer Gestures

### LINKED ISSUE #5741: Fullscreen mode: swiping down to show the menu interferes with "swipe down" gesture
###### Reproduction Steps

1. Set Setting > reviewing > fullscreen mode to: hide system bars and answer buttons
2. Set Settings > gestures > swipe down to: answer button 2 or anything else
3. Review a card in fullscreen, try to show the menu by swiping down from the edge of your phone

###### Expected Result

The menu is shown

###### Actual Result

The menu is shown for a millisecond, then the defined swipe gesture is activated (in my case: mark the card with button 2 and go to the next card)

Possible solution: swipe gestures should not be activated when swiping from the edge of the screen.

This issue is the result of pull request https://github.com/ankidroid/Anki-Android/pull/5628 which solved an issue that I created myself (https://github.com/ankidroid/Anki-Android/issues/5415) :man_facepalming: 

Obviously the reason for the margins in fullscreen was to prevent the swipe down gesture to interfere with the show menu gesture. But since those margins lead to a misalignment of card contents and touch area, it is good that the margins got removed with pull request 5628.

So there should be another way to prevent the "show menu gesture" with the user defined swipe down gesture.

###### Debug info
AnkiDroid Version = 2.10alpha33

Android Version = 9

ACRA UUID = 1f29f610-893a-403f-83d5-25a26ba97fe0

###### Research
*Enter an [ x ] character to confirm the points below:*

[ x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[X  ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[ x ] I have searched for similar existing issues here and on the user forum

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description
This fixes two issues, both when Fullscreen (No Buttons) is active.

Firstly: A swipe down from the status bar first triggered the Anki UI, and then was registered as a fling. This fling assumes "I'm a fling when the Anki UI is active, I should close the Anki UI and process a gesture".

Secondly: When drawing from the top and the whiteboard, the Anki UI is shown. The Reviewer logic assumes "I'm handling a touch event when the Anki UI is shown", this stops the draw event, and starts a fling from wherever the cursor is at the current moment, which closes the Anki UI and performs a gesture.

## Fixes
Fixes #5741

## Approach

* We define a margin, and assume it's not a gesture if it's within the margin of the edge of the screen.
* A whiteboard draw event continues after it shows the Anki UI.

## How Has This Been Tested?

On my Android

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

## Review Goals

* Grill me on the hardcoded 24dip for the status bar, I felt it was fine given that we previously used a hardcoded value of 20dip.
* Should the whiteboard also have a margin? If the user draws from the top of the screen, do they want to draw, or get the navigation bar?

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/Whiteboard.java [hunk 1]
```diff
@@ -28,6 +28,8 @@
 import android.graphics.Point;
 import android.graphics.PointF;
 import androidx.core.content.ContextCompat;
+import timber.log.Timber;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/Whiteboard.java]: Codacy Issue found: [Avoid unused imports such as 'timber.log.Timber'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5261420)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5261420&bid=17237321)

File: AnkiDroid/src/main/java/com/ichi2/anki/Whiteboard.java [hunk 2]
```diff
@@ -36,6 +35,8 @@
 import java.lang.ref.WeakReference;
 import java.util.Stack;
 
+import androidx.core.content.ContextCompat;
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/Whiteboard.java]: just noting the import move is gratuitous

**Comment by david-allison-1** [comment]
> This all seems fine to me, I don't use gestures and not many do I think so 20 vs 24 doesn't matter to me. Luckily the alpha group is heavily populated with exactly the people that do use them so you'll get feedback quickly. You should definitely canvas the linked issues to rope the issue loggers in for feedback, then basically works for me - approving now but if you could move the import back just to minimize diff that's always good practice

Based on feedback, it'd be better to have the Whiteboard not draw on edges.

I'll shelve this and will get back to it later in the week, happy to leave it on "Awaiting Changes". This is a complex area of the code, and there's a few current bugs that it's brought into the limelight (Dictionary Lookup fails due to the clipboard stating it doesn't have text, there's a "LongClick" gesture, which is never set, and the code to trigger it is buggy). 

@mikehardy I feel it would be best to add tests for the changes that I make here, and think it's best to extract the touch logic into a separate class (mostly the dispatch logic for either Whiteboard, WebView, Gestures, Dictionary Lookup or UI Hiding). Would this be feasible to get through a review here assuming it's tested and good quality, or would you prefer a later refactoring & testing PR?

**Comment by mikehardy** [comment]
I never mind reviewing refactors and added testing actually. I greatly appreciate when they are separated into their own commits so it's really easy to see that a refactor doesn't change actual logic, but to me those fun to read normally, I can see the code getting better...

**Comment by Arthur-Milchior** [comment]
I've used gesture for more than a year. i just discovered full screen. Really useful when I review images. And it's really hard to get the buttons because gesture often captures my request for the buttons to be back. Any margin that allows gesture to be deactivated when the gesture starts on the border of the screen will be really appreciated !

**Comment by mikehardy** [comment]
hey @david-allison-1 was there more work on this one? I'd put it in 2.10 if you thought it was good to go. Maintaining that release label for tracking until I hear back, it'll assure I follow up either way

**Comment by david-allison-1** [comment]
> hey @david-allison-1 was there more work on this one? I'd put it in 2.10 if you thought it was good to go. Maintaining that release label for tracking until I hear back, it'll assure I follow up either way

Hmm... it's not a buggy PR, and improves the experience, but it's not perfect and I'd like to work on it a little more. 

Would it be OK to merge, but keep the issue open?

**Comment by mikehardy** [comment]
I'd rather wait than merge+rework, it will likely be pickable and either way release cadence will be pretty quick, I don't intend to let any improvements sit long before wide release going forward

**Comment by mikehardy** [comment]
Moving to 2.11 tag to indicate it's not gating 2.10, if it's pickable, excellent but if not to fuss - 2.11 won't be too far along

**Comment by mikehardy** [comment]
How do you feel about this now? Can merge or not, I'm generally inclined to merge unless you have a definite thing you want to do

**Comment by david-allison-1** [comment]
It's been around for too long, the code's not bad, and it's a bugfix. 

I'd like to mark this area as 'get back to it'

Perfect is the enemy of good. Let's get it in there and improve the area later on.

**Comment by mikehardy** [comment]
boom! 2.11 will have it

**Pull request merged**